lib/ap/internal/dir.h

```diff
@@ -42,50 +35,50 @@ typedef	unsigned long ulong;
 typedef union
 {
 	char	clength[8];
-	vlong	vlength;
+	int64_t	vlength;
 	struct
 	{
-		long	hlength;
-		long	length;
+		int32_t	hlength;
+		int32_t	length;
 	};
 } Length;
 
 typedef
 struct Qid
 {
-	uvlong	path;
+	uint64_t	path;
 	uint32_t	vers;
-	uchar	type;
+	unsigned char	type;
 } Qid;
 
 typedef
 struct Dir {
 	/* system-modified data */
 	uint16_t	type;	/* server type */
-	uint	dev;	/* server subtype */
+	uint32_t	dev;	/* server subtype */
 	/* file data */
 	Qid	qid;	/* unique id from server */
 	uint32_t	mode;	/* permissions */
 	uint32_t	atime;	/* last read time */
 	uint32_t	mtime;	/* last write time */
-	vlong	length;	/* file length: see <u.h> */
+	int64_t	length;	/* file length: see <u.h> */
 	char	*name;	/* last element of path */
 	char	*uid;	/* owner name */
 	char	*gid;	/* group name */
 	char	*muid;	/* last modifier name */
 } Dir;
 
 void	dirtostat(struct stat *, Dir*, Fdinfo*);
-uint	convM2D(unsigned char*, uint, Dir*, char*);
-uint	convD2M(Dir*, uchar*, uint);
+uint32_t	convM2D(unsigned char*, uint32_t, Dir*, char*);
```

Comment:
This is one case where specifying the big length makes no sense to me. It will fit in anything.